controller/control.go

```diff
@@ -234,6 +234,11 @@ func (c *Controller) Start(addresses ...string) error {
 
 func (c *Controller) WriteAt(b []byte, off int64) (int, error) {
 	c.RLock()
+	if off < 0 || off+int64(len(b)) > c.size {
```

Comment:
`bin/longhorn` I meant.

if `make` failed, you won't get new `bin/longhorn` binary with your latest change. So you may need to run `make build` to get binary, and skip all the tests to get the latest binary.